db/models/model.js

```diff
@@ -1,34 +1,60 @@
 const db = require("../connection");
-// const { articleData } = require("../data/test-data");
 
 const fetchTopicsFromDatabase = () => {
   return new Promise((resolve, reject) => {
-    db.query("SELECT * FROM topics;")
-      .then((queryResult) => {
-        resolve(queryResult.rows);
-      })
-      .catch((error) => {
-        reject(error);
-      });
+    db.query("SELECT * FROM topics;").then((queryResult) => {
+      resolve(queryResult.rows);
+    });
   });
 };
 
+function fetchAllArticlesFromDataBase() {
+  return new Promise((resolve, reject) => {
+    db.query(
+      "SELECT article_id, title, topic, author, created_at, votes, article_img_url FROM articles ORDER BY created_at DESC;"
+    ).then(async (queryResult) => {
+      const articles = queryResult.rows;
+
+      for (const article of articles) {
+        const articleId = article.article_id;
+
+        const commentCount = await getCommentCountForArticle(articleId);
+
+        article.comment_count = commentCount;
+      }
+
+      resolve(articles);
+    });
+  });
+}
+function getCommentCountForArticle(articleId) {
+  return new Promise((resolve, reject) => {
+    db.query(
```

Comment:
I would have just added the comment count to the query in the function above and not had a separate function to getCommentCount